### ISSUE TITLE
Prettier does not seem to like a comment that starts with two colons

### ISSUE THREAD

**Issue by benjie** [prettier/prettier#6102]
**Prettier 1.17.0**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBDAFAHSgAhwenxySWzwEZsBKAbmxABoQIAHGAS2gGdlRUAnfhADuABQEIeKVADcI7ACaMQAI36owAazgwAyiw3soAc2Qx+AVzhMAFjAC2AGwDqN9vC4GwcXZPfsZdwBPZHAuHiYjLjh+GFF1Y3tUZAAzVEdopgArLgAPACF1LR1dVHs4ABkjOFT0zJAc3N0jY0c4AEULCHhajOsQA35o-lCVVBUgx2hlFn4jGGdFGBtkAA4ABiZZiGjndRZQ2bhhmRqmAEcu+HjWKTQuAFooODgFV+V+OEv2T-jUROSSDSfSY0Xs7DMln6XBabU63RqSHMViYMHGiwUy2QACZUep2I4WgBhCD2JKhY4AVmUFmiABVxlJgfUFBAwL1ogBfTlAA)
```sh
--parser babylon
--single-quote
--trailing-comma es5
```

**Input:**
```jsx
a(
  // ::
  1
);

```

**Output:**
```jsx
Error: Comment "::" was not printed. Please report this error!
    at https://prettier.io/lib/standalone.js:16137:13
    at Array.forEach (<anonymous>)
    at ensureAllCommentsPrinted (https://prettier.io/lib/standalone.js:16135:15)
    at coreFormat (https://prettier.io/lib/standalone.js:16186:3)
    at format (https://prettier.io/lib/standalone.js:16427:73)
    at formatWithCursor (https://prettier.io/lib/standalone.js:16443:12)
    at https://prettier.io/lib/standalone.js:30799:15
    at Object.format (https://prettier.io/lib/standalone.js:30807:12)
    at formatCode (https://prettier.io/worker.js:234:21)
    at handleMessage (https://prettier.io/worker.js:185:18)
```

**Expected behavior:**

---

This is a minimal reproduction, this issue was triggered by this file:

https://github.com/graphile/graphile-engine/blob/4ba5eb30eacd3bec8e1dccc10ff6bbdb6c765794/packages/graphile-build-pg/src/plugins/PgTypesPlugin.js

**Comment by purge** [comment]
i'm seeing the same when parser is set to `babel` or `babel-flow` in the playground.

**Comment by jordwalke** [comment]
Seeing this as well in the following example (playground, latest version):

```
module.exports = {};

module.exports = ((module.exports /*:: : a*/) /*:: :b */);
```

**Comment by warpr** [comment]
Also happens with just a single colon:

**Prettier 2.0.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAzAlgGzjUwF5MAKcgeQCMArOMGAOgDcBDfAVyPIAd2AczjM8hNAEoA9ACpMSTOygBPTDKkTMs+ZgCCAJ33tlAHgBiBOAFkIAEzj4AfGo0BuADpQQAGhAQ+MLjoyKDshhAA7gAKYQhoyCCcEcbxvjRGYADWcDAAygJguFCCyDD6PL4AFjAAtvgA6pW48GgFcLlxzbiszcoJYGipIEVocPowUUaCNezI2JyjvnRoAB4AQhnZeew1cAAyRXBzC3BLq7lFgoQAilwQ8Mf4iyAC+qP6CTTsNA4+L-pFGD1XC2GCVZAADgADL4+PoIKN6kY+Ak4UQxqwjr4AI53eCTfzxFDsNAAWigcDg9lsf30cFxuDpkyEM0ez1GNVwpXKpxAaEuNzxRyQ8yevJg32BoPBSAATL4yuwCJcAMIQGqslBEACsfy4owAKt8iaLnqweABJKD2WC5MAAgK6a25GDKQhsuAAX09QA)
```sh
--parser babel
```

**Input:**
```jsx
const files = ((Object.values(page.files)/*: any */) /*: Array<FileModel> */);

```

**Output:**
```jsx
ensureAllCommentsPrinted/<@https://prettier.io/lib/standalone.js:17105:15
ensureAllCommentsPrinted@https://prettier.io/lib/standalone.js:17103:17
coreFormat@https://prettier.io/lib/standalone.js:17151:29
format@https://prettier.io/lib/standalone.js:17369:77
formatWithCursor@https://prettier.io/lib/standalone.js:17385:14
withPlugins/<@https://prettier.io/lib/standalone.js:32887:14
format@https://prettier.io/lib/standalone.js:32896:14
formatCode@https://prettier.io/worker.js:117:21
handleMessage@https://prettier.io/worker.js:68:18
self.onmessage@https://prettier.io/worker.js:38:14

```

**Expected behavior:**

```js
const files = ((Object.values(page.files) /*: any */) /*: Array<FileModel> */);
```

**Comment by mrtnzlml** [comment]
I just encountered the same problem when testing the new Rust Compiler for Relay which generates the following code:

```js
const node = " … ";
module.exports = ((node/*: any*/)/*: Mutation<
  ProductEditFormMutation$variables,
  ProductEditFormMutation$data,
>*/);
```

Results in the following error:

```text
Error: Comment ": Mutation<
  ProductEditFormMutation$variables,
  ProductEditFormMutation$data,
>" was not printed. Please report this error!
    at Object.ensureAllCommentsPrinted (https://prettier.io/lib/standalone.js:1:218494)
    at qD (https://prettier.io/lib/standalone.js:1:225762)
    at JD (https://prettier.io/lib/standalone.js:1:228265)
    at Object.formatWithCursor (https://prettier.io/lib/standalone.js:40:40390)
    at formatCode (https://prettier.io/worker.js:141:21)
    at handleMessage (https://prettier.io/worker.js:78:26)
    at self.onmessage (https://prettier.io/worker.js:36:14)
```

https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEUIBM6YC8mAOiJoGQEpIA3CVALZ4CuANnAHRwAeADhACcYaIpgAUYnPgD0AKiSYAhlACes6QEo5CgLLMYimAEtoAHnqZMABQEswMAKK4jMAGKCGeg8egASAG6KAkaKAEbsaAA0Fta2uMz2Ti7uAp76hiZQvriGitFQAHzqGnRQIJEgELw+UGjIoEG2AO5WQQh1KIqsTYoqdRWhAopgANZwMADKigxwADJGUHDIAGZdaHADQ6PjE7zDCwDmyDACzBsgcAyhcLj4uLPKB8yKB3ApDIbGUEed+hDlIAAFjAGKwAOqAlxwNB7MBwCbtFxGfwuFTIcBofogBbrIQ2F4fFZrc4AKzQ3Amh3YAEVmBB4ETWOsKnsBLj0aEwnBWADeMFYGCjLgYIDkAAOAAMLNs6zBQ146L50LgAn8SwqAEc6fAbFUOiBFGgALSLG43AECOBaoyW-EHQlIVZM87rBhGY6nF1UuC0+lLR3EioGUKC4WipAAJiDQyMrEOAGEIAwHRc0ABWAHMdYAFTCHSdzJA-jOAEkoPhYBMwMFqgBBcsTGAqdiM9YAXzbQA

**Comment by thorn0** [comment]
Fixed by #13687
